client/state/themes/reducer.js

```diff
@@ -483,7 +483,10 @@ export const themeFilters = withSchemaValidation( themeFiltersSchema, ( state =
 export function recommendedThemes( state = {}, action ) {
 	switch ( action.type ) {
 		case RECOMMENDED_THEMES_FETCH:
-			return { ...state, [ action.filter ]: { isLoading: true, themes: [] } };
+			return {
+				...state,
+				[ action.filter ]: { isLoading: true, themes: state[ action.filter ]?.themes ?? [] },
```

Comment:
It makes sense to keep existing data while fetching new data. Displaying existing stuff for the user is 👌  I think we should have this in all three tabs -- not just recommended. But that'd be a follow-up.